### PR TITLE
Add IoT Hub reading function and Azure table reading function

### PR DESCRIPTION
**Pull request by ZhijunZhao**
@zikalino @xscript @aovechki @shizn @reyang

### CODE REVIEW

**Comment by ZhijunZhao** [comment]
Next step, I'll try to consolidate the gulp files of blink sample and azure blink sample.

File: azure-table.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            if (error) {
+              console.error('Fail to read messages:\n' + error);
+              setTimeout(readNewMessage, 0);
+              return;
+            }
+
+            // result.entries contains entities matching the query
+            if (result.entries.length > 0) {
+              for (var i = 0; i < result.entries.length; i++) {
+                console.log('[Azure table] Read message #' + ++messageCount + ': ' + result.entries[i].message['_'] + '\n');
```
Comment (xscript) [inline azure-table.js]: Can we move `++messageCount` to a separate line for better readability?

Comment (xscript) [inline azure-table.js]: table -> Table?

File: azure-table.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      } else {
+        console.error('ERROR: Fail to get connection string of Azure Storage account.')
+      }
+    } else {
+      console.error('ERROR: No output when getting connection string of Azure Storage account.');
+    }
+  });
+}
+
+var callback = function (deferred) {
```
Comment (xscript) [inline azure-table.js]: Can we use more meaningful name instead of callback? Such as `actionOnExit` or `callbackOnExit`.

File: iot-hub.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+var EventHubClient = require('azure-event-hubs').Client;
```
Comment (aovechki) [inline iot-hub.js]: please add 'use strict' and make sure eslint is clean

File: azure-table.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  // Wait 5 more sends so that Azure function has the chance to process sent messages.
```
Comment (aovechki) [inline azure-table.js]: >sends [](start = 17, length = 5)

sends->seconds

File: azure-table.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      var connStr = JSON.parse(stdout).connectionString;
+      if (connStr) {
+        var tableService = storage.createTableService(connStr);
+        var condition = 'PartitionKey eq ? and RowKey gt ? ';
+        var tableName = 'DeviceData';
+        var timestamp = moment.utc().format('hhmmssSSS');
+
+        var messageCount = 0;
+
+        function readNewMessage() {
```
Comment (aovechki) [inline azure-table.js]: >readNewMessage [](start = 17, length = 14)

readNewMessage->readNewMessages

File: azure-table.js (hunk 2 above)
Comment (ZhijunZhao) [inline azure-table.js]: Agree. Let me try resolving this in next round of PRs.

**Pull request merged**